op-service/logfilter/logfilter.go

```diff
@@ -0,0 +1,222 @@
+	return buildFilter(false, filters...)
+}
+
+// ============================================================================
+// Filter Handler Types
+// ============================================================================
+
+type Filter func(ctx context.Context, lvl slog.Level) bool
+
+// FilterHandler interface that can be used with the log filtering system
+type FilterHandler interface {
+	logmods.Handler
+	// config parameter can be created with logfilter.DefaultShow(<filters...>) or logfilter.DefaultMute(<filters...>)
+	Set(filter Filter)
+}
+
+// filterHandler implements the Handler interface
+type filterHandler struct {
+	slog.Handler
+	filter atomic.Pointer[Filter]
+	attrs  map[string]slog.Value
+	group  string
+}
+
+var _ FilterHandler = &filterHandler{}
+
+func (f *filterHandler) Set(filter Filter) {
+	f.filter.Store(&filter)
+}
+
+func (f *filterHandler) Unwrap() slog.Handler {
+	return f.Handler
+}
+
+func (f *filterHandler) Enabled(ctx context.Context, lvl slog.Level) bool {
+	// First check if the log is enabled by the inner handler, because these are less expensive to evaluate
+	if !f.Handler.Enabled(ctx, lvl) {
+		return false
+	}
+	filter := f.filter.Load()
+	if filter == nil {
+		return true
+	}
+	// Dynamically load the context with the attrs
+	for k, v := range f.attrs {
+		ctx = log.AddLogAttrToContext(ctx, k, v.LogValuer())
+	}
+	result := (*filter)(ctx, lvl)
+	return result
+}
+
+func (h *filterHandler) WithAttrs(as []slog.Attr) slog.Handler {
+	attrs := make(map[string]slog.Value)
+	// copy h.attrs to attrs
+	for k, v := range h.attrs {
+		attrs[k] = v
+	}
+	for _, a := range as {
+		var key string
+		if h.group != "" {
+			key = h.group + "." + a.Key
+		} else {
+			key = a.Key
+		}
+		attrs[key] = a.Value
+	}
+	handler := filterHandler{Handler: h.Handler.WithAttrs(as), filter: atomic.Pointer[Filter]{}, attrs: attrs, group: h.group}
+	handler.filter.Store(h.filter.Load())
+	return &handler
+}
+
+func (h *filterHandler) WithGroup(name string) slog.Handler {
+	attrs := make(map[string]slog.Value)
+	// copy h.attrs to attrs
+	for k, v := range h.attrs {
+		attrs[k] = v
+	}
+	handler := filterHandler{Handler: h.Handler.WithGroup(name), filter: atomic.Pointer[Filter]{}, attrs: attrs, group: name}
+	handler.filter.Store(h.filter.Load())
+	return &handler
+}
+
+// Handler Wrapper
+// WrapFilterHandler wraps a slog.Handler with filtering capabilities
+var _ logmods.HandlerMod = WrapFilterHandler
+
+func WrapFilterHandler(h slog.Handler) slog.Handler {
+	handler := &filterHandler{
+		Handler: h,
+		filter:  atomic.Pointer[Filter]{},
+	}
+	return handler
+}
+
+// ============================================================================
+// Selector Functions
+// ============================================================================
+
+type ComparableLogValuer interface {
+	slog.LogValuer
+	comparable
+}
+
+func SelectorFn[T ComparableLogValuer](key string, value T) Selector {
```

Comment:
Can you add a doc comment to say that the log-context attribute naming is being used to do the selecting?